bots/csr/src/main/java/org/openjdk/skara/bots/csr/CSRBot.java

```diff
@@ -58,32 +64,46 @@ private String describe(PullRequest pr) {
 
     @Override
     public void run(Path scratchPath) {
-        for (var pr : repo.pullRequests()) {
+        var prs = repo.pullRequests();
+        for (var pr : prs) {
+            if (!cache.needsUpdate(pr)) {
+                continue;
+            }
+
             log.info("Checking CSR label for " + describe(pr) + "...");
             if (pr.labels().contains(CSR_LABEL)) {
-                var issue = org.openjdk.skara.vcs.openjdk.Issue.fromString(pr.title());
-                if (issue.isEmpty()) {
-                    log.info("No issue found in title for " + describe(pr));
-                    continue;
-                }
-                var jbsIssue = project.issue(issue.get().id());
-                if (jbsIssue.isEmpty()) {
-                    log.info("No issue found in JBS for " + describe(pr));
-                    continue;
-                }
+                hasCSRLabel.add(pr.id());
+            }
```

Comment:
Perhaps check if the CSR_LABEL is gone but the PR is in hasCSRLabel, in case it was manually removed?

Reply:
Good idea, will fix!

Reply:
Fixed in 38429c07fbd20d66ffc6d4eda0f0ddee9c7397a0